finagle-netty4/src/test/scala/com/twitter/finagle/netty4/Netty4SslTest.scala

```diff
@@ -0,0 +1,111 @@
+
+import java.net.{InetAddress, InetSocketAddress, SocketAddress}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import com.twitter.finagle
+import com.twitter.finagle.Stack.Params
+import com.twitter.finagle._
+import com.twitter.finagle.client.{StackClient, StdStackClient, Transporter}
+import com.twitter.finagle.dispatch.{SerialClientDispatcher, SerialServerDispatcher}
+import com.twitter.finagle.param.Label
+import com.twitter.finagle.ssl.client.{SslClientConfiguration, SslClientEngineFactory}
+import com.twitter.finagle.ssl.server.{SslServerConfiguration, SslServerEngineFactory}
+import com.twitter.finagle.ssl.{ClientAuth, Engine}
+import com.twitter.finagle.transport.Transport
+import com.twitter.finagle.transport.Transport.ServerSsl
+import com.twitter.util.{Await, Future}
+import io.netty.channel.ChannelPipeline
+import io.netty.handler.codec.string.{StringDecoder, StringEncoder}
+import io.netty.handler.codec.{DelimiterBasedFrameDecoder, Delimiters}
+import io.netty.handler.ssl.SslContextBuilder
+import io.netty.handler.ssl.util.SelfSignedCertificate
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.concurrent.{Eventually, IntegrationPatience}
+import org.scalatest.junit.JUnitRunner
+
+object Netty4SslTest {
+  case class Client(clientCert: SelfSignedCertificate, serverCert: SelfSignedCertificate,
+                    params: Params = Params.empty,
+                    stack: Stack[ServiceFactory[String, String]] = StackClient.newStack)
+    extends StdStackClient[String, String, Client] {
+
+    override protected type In = String
+    override protected type Out = String
+
+    override protected def newTransporter(addr: SocketAddress): Transporter[String, String] =
+      Netty4Transporter.raw[String, String](pipeline => {
+        pipeline.addLast(new DelimiterBasedFrameDecoder(8192, Delimiters.lineDelimiter(): _*))
+        pipeline.addLast(new StringDecoder())
+        pipeline.addLast(new StringEncoder())
+      }, addr, params)
+
+    override protected def newDispatcher(transport: Transport[String, String]): Service[String, String] =
+      new SerialClientDispatcher(transport)
+
+    override protected def copy1(stack: Stack[ServiceFactory[String, String]], params: Params) =
+      copy(clientCert, serverCert, params, stack)
+  }
+}
+
+@RunWith(classOf[JUnitRunner])
+class Netty4SslTest extends FunSuite with Eventually with IntegrationPatience {
+  object StringServerInit extends (ChannelPipeline => Unit) {
+    def apply(pipeline: ChannelPipeline): Unit = {
+      pipeline.addLast("line", new DelimiterBasedFrameDecoder(100, Delimiters.lineDelimiter(): _*))
+      pipeline.addLast("stringDecoder", new StringDecoder(UTF_8))
+      pipeline.addLast("stringEncoder", new StringEncoder(UTF_8))
+    }
+  }
+
+  val serverCert = new SelfSignedCertificate()
+  val clientCert = new SelfSignedCertificate()
+  val allocator = io.netty.buffer.UnpooledByteBufAllocator.DEFAULT
+
+  val server = {
```

Comment:
how about making a `class Ctx` and shoving this server in there?  that way we can write our tests like:

```
test("some test stuff") {
  val ctx = new Ctx()
  import ctx._
  server.blahblahblah()
  ....
}
```

and each test will have its own server, but we can write it as if the server was some shared resource.